### PR TITLE
[HDInsight] Support new azure monitor feature

### PR DESCRIPTION
**Pull request by aim-for-better**
## Description

This PR add three cmdlets to support new azure monitor feature

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by aim-for-better** [comment]
This is for Draft. Thanks~

**Comment by msJinLei** [comment]
@aim-for-better Is the pr ready for review?

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [inline src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs:30]: Please use appropriate parameter set names.

File: src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [inline src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs:32]: Please use appropriate parameter set names.

File: src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [inline src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightAzureMonitorCommand.cs:30]: Please use appropriate parameter set names.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by aim-for-better** [comment]
> @aim-for-better Is the pr ready for review?

Thanks @msJinLei , this PR is not ready, maybe there is still small changes.

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs:30]: > Please use appropriate parameter set names.

Hi @msJinLei , can we use this string? The other existing commands also use the parameter set: SetByNameParameterSet.
This means customer need to provide cluster name instead of providing resource id or object via pipeline.
We want to keep the same code style......

**Comment by aim-for-better** [comment]
Hi @msJinLei, the PR is ready now, could you please help review? Thanks

File: src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs:32]: > Please use appropriate parameter set names.

Hi @msJinLei, there are many other service also are using this parameter set. Can we keep this?

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [reply src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs:30]: You can name the parameter set as, for example, ByNameParameterSet = "ByNameParameterSet". SetByNameParameterSet leads to misunderstanding.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by aim-for-better** [comment]
Hi @msJinLei thanks for your comment. The only concern is that changing it will break up our code style, and there are many services are using this parameterset. Hoping you can consider this waiting for your final decision thank you~

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [reply src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs:30]:
```
src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightGatewayCredential.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Set-AzHDInsightGatewayCredential.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightGatewayCredential.md:Parameter Sets: SetByNameParameterSet
```
 "ByNamePrameterSet" seems to better in your case. You don't need to change all the cmdlets. Just change the cmdlets included in your PR.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by msJinLei** [comment]
The PR depends on  #15172

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs]: Changed per your comments. Thanks

File: src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs]: changed to bynameparameterset

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [inline src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs:30]: My proposal is to remove all the prefixes "Set" as you cmdlet verb is not "Set".
```
        private const string ByNameParameterSet = "ByNameParameterSet";
        private const string ByResourceIdParameterSet = "ByResourceIdParameterSet";
        private const string ByInputObjectParameterSet = "ByInputObjectParameterSet";
```

Another option is to use cmdlet verb as the prefix.
```
        private const string DisableByNameParameterSet = "DisableByNameParameterSet ";
        private const string DisableByResourceIdParameterSet = "DisableByResourceIdParameterSet";
        private const string DisableByInputObjectParameterSet = "DisableByInputObjectParameterSet";
```
The principle is the variable name should be consistent with its meaning and usage.

File: src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [inline src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs:32]: My proposal is to remove all the prefixes "Set" as you cmdlet verb is not "Set".
```
        private const string ByNameParameterSet = "ByNameParameterSet";
        private const string ByResourceIdParameterSet = "ByResourceIdParameterSet";
        private const string ByInputObjectParameterSet = "ByInputObjectParameterSet";
```

File: src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightAzureMonitorCommand.cs
Comment (msJinLei) [inline src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightAzureMonitorCommand.cs:29]: My proposal is to remove all the prefixes "Set" as you cmdlet verb is not "Set".
```
        private const string ByNameParameterSet = "ByNameParameterSet";
        private const string ByResourceIdParameterSet = "ByResourceIdParameterSet";
        private const string ByInputObjectParameterSet = "ByInputObjectParameterSet";
```

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/GetAzureHDInsightAzureMonitorCommand.cs:29]: > 

fixed. Thanks~

File: src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs]: > My proposal is to remove all the prefixes "Set" as you cmdlet verb is not "Set".
> 
> ```
>         private const string ByNameParameterSet = "ByNameParameterSet";
>         private const string ByResourceIdParameterSet = "ByResourceIdParameterSet";
>         private const string ByInputObjectParameterSet = "ByInputObjectParameterSet";
> ```

Fixed. Thanks

File: src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs]: > My proposal is to remove all the prefixes "Set" as you cmdlet verb is not "Set".
> 
> ```
>         private const string ByNameParameterSet = "ByNameParameterSet";
>         private const string ByResourceIdParameterSet = "ByResourceIdParameterSet";
>         private const string ByInputObjectParameterSet = "ByInputObjectParameterSet";
> ```
> 
> Another option is to use cmdlet verb as the prefix.
> 
> ```
>         private const string DisableByNameParameterSet = "DisableByNameParameterSet ";
>         private const string DisableByResourceIdParameterSet = "DisableByResourceIdParameterSet";
>         private const string DisableByInputObjectParameterSet = "DisableByInputObjectParameterSet";
> ```
> 
> The principle is the variable name should be consistent with its meaning and usage.

Fixed. changed to the second option. Thanks~

**Review by msJinLei — APPROVED** [review]

**Comment by aim-for-better** [comment]
The CI passed

**Pull request merged**
